lab-spencer/route/seahawk-router.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const Seahawk = require('../model/seahawk.js');
+
+let seahawkRouter = module.exports = new Router();
+
+seahawkRouter.post('/api/seahawks', jsonParser, (req, res, next) => {
+  new Seahawk(req.body)
+    .save()
+    .then(seahawk => res.json(seahawk))
+    .catch(err => {
+      if(err.name === 'ValidatorError' || err.name === 'ValidationError')
+        return res.sendStatus(400);
+      next(err);
+    });
+});
+
+seahawkRouter.get('/api/seahawks', (req, res, next) => {
+  Seahawk.find({})
+    .then(seahawk => res.json(seahawk))
+    .catch(next);
+});
+
+seahawkRouter.get('/api/seahawks/:id', (req, res, next) => {
+  Seahawk.findById(req.params.id)
+    .then(seahawk => res.json(seahawk))
+    .catch(next);
+});
+
+seahawkRouter.put('/api/seahawks/:id', jsonParser, (req, res, next) => {
+  Seahawk.update({_id: req.params.id}, {name: req.body.name, height: req.body.height, weight: req.body.weight, position: req.body.position, picture: req.body.picture})
```

Comment:
Mongoose is a bit weird in that it returns the old object instead of the new one, I'm not quite sure why (I have done some searching on the matter, and came up blank).

As for your put, I'll just show you how I would do it and I'm sure you can pick up the things.

```js
seahawkRouter.put('/api/seahawks/:id', jsonParser, (req, res, next) => {
  Seahawk.findByIdAndUpdate(req.params.id, req.body, { new: true })
    .then((newSeahawk) => res.json(newSeahawk))
    .catch(next);
});
```